packages/block-library/src/navigation-area/edit.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { store as coreStore } from '@wordpress/core-data';
+import {
+	MenuGroup,
+	MenuItemsChoice,
+	PanelBody,
+	SelectControl,
+	ToolbarDropdownMenu,
+	ToolbarGroup,
+} from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import { BlockControls, InspectorControls } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import InnerBlocks from './inner-blocks';
+import PlaceholderPreview from '../navigation/edit/placeholder/placeholder-preview';
+
+function NavigationAreaBlock( { attributes, setAttributes } ) {
+	const { area } = attributes;
+
+	const { navigationAreas, hasResolvedNavigationAreas } = useSelect(
+		( select ) => {
+			const { getEntityRecords, hasFinishedResolution } = select(
+				coreStore
+			);
+			return {
+				navigationAreas: getEntityRecords( 'root', 'navigationArea' ),
+				hasResolvedNavigationAreas: hasFinishedResolution(
```

Comment:
We could do `&& navigationAreas.length` to prevent pretending everything is fine when there are no areas